Restore RGElementDefinition>>realParent's behavior of not setting the parent instance variable

akgrant43 opened the pull request:
`#parent` answers the receiver's model parent, which may be `nil` if not set, while `#realParent` answers the actual instance in the system.

fix: pharo-project#18440

akgrant43 committed:
Restore RGElementDefinition>>realParent's behavior of not setting the parent instance variable.

`#parent` answers the receiver's model parent, which may be `nil` if not set, while `#realParent` answers the actual instance in the system.

fix: pharo-project#18440

Ducasse commented:
Hi alistair 
I'm back from holidays while others are leaving so I will try to go regularly over issues. 
I saw that the following test is related to your changes and breaking. 

S

akgrant43 committed:
RGMethodDefinitionTest>>testAsFullMethodDefinition should work from the class's full ring definition.

Previously #testAsFullMethodDefinition retrieved the method's full ring definition and then assumed that the class contained the full ring definition.  This relied on the incorrect behaviour of RGMethodDefinition>>realParent setting the model parent.  It also clearly contradicts the comment in CompiledMethod>>asFullRingDefinition which states that "the full conversion does not happen at the level of the class. If you need that request asFullRingDefinition to the class".

fix: #18444

jecisc committed:
Merge branch 'Pharo14' into 18440-RGElementDefinitionrealParent-corrupts-the-model

MarcusDenker committed:
Merge branch 'Pharo14' into 18440-RGElementDefinitionrealParent-corrupts-the-model

MarcusDenker committed:
Merge branch 'Pharo14' into 18440-RGElementDefinitionrealParent-corrupts-the-model
